### PR TITLE
refactor file pipeline flow

### PR DESCRIPTION
**Pull request by mirrir0**

### CODE REVIEW

File: backend/crawler/docingest.py [hunk 1]
```diff
@@ -40,44 +38,36 @@ def __init__(
...
-        self.logger.info("Successfully got file and metadata")
+        self.logger.info("collecting filepath metadata")
+
+        try:
+            filepath, metadata = self.add_file_from_url_nocall(url)
+            self.logger.info("Successfully got file and metadata")
+        except Exception:
+            self.logger.error("unable to get file and metadata")
+            pass
+
```
Comment (mirrir0) [inline backend/crawler/docingest.py:59]: all of the rest of this was dead code

File: backend/crawler/docingest.py [hunk 2]
```diff
@@ -40,44 +38,36 @@ def __init__(
...
         file_metadata = self.get_metadata_from_file(filepath, metadata.get("doctype"))
         self.logger.info("Attempted to get metadata from file, adding to main source.")
         # FIXME :
         # metadata.update(file_metadata)
-        return (filepath, metadata)
+
+        def cleanup():
+            os.remove(filepath)
+
+        return (filepath, metadata, cleanup)
```
Comment (mirrir0) [inline backend/crawler/docingest.py:70]: the cognitive load for cleaning up the temporary path should be in the function it comes from, not the place it happens

File: backend/crawler/docingest.py [hunk 3]
```diff
@@ -292,107 +283,3 @@ def get_blake2_str(
 
     def infer_metadata_from_path(self, filepath: Path) -> dict:
         return {"title": filepath.stem, "doctype": filepath.suffix}
```
Comment (mirrir0) [inline backend/crawler/docingest.py:285]: got rid of dead non-reference code. we dont need to hoard

File: backend/routing/file_controller.py [hunk 4]
```diff
@@ -157,20 +163,22 @@ async def get_all_files(
...
+    #     limit_offset: LimitOffset,
+    #     request: Request,
+    #     reprocess_all: bool = False,
+    # ) -> list[FileSchema]:
+    #     """List files."""
+    #     results = await files_repo.list()
+    #     type_adapter = TypeAdapter(list[FileSchema])
+    #     for file in results:
+    #         process_file_raw(file, files_repo, request.logger, reprocess_all)
+    #     return type_adapter.validate_python(results)
```
Comment (mirrir0) [inline backend/routing/file_controller.py:181]: not quite dead, just needs to be moved

File: backend/routing/file_controller.py [hunk 5]
```diff
@@ -328,91 +354,80 @@ async def process_file_raw(
...
-        logger.info(response_code)
-        logger.info(response_message)
-        newobj = files_repo.update(obj)
+            except Exception as e:
+                raise Exception("Failure in adding document to vector database", e)
 
-        await files_repo.session.commit()
+        while True:
+            try:
+                match current_stage:
```
Comment (mirrir0) [inline backend/routing/file_controller.py:402]: holy molly please never use `if x=a else if x=b else .... ` again

File: backend/routing/file_controller.py [hunk 6]
```diff
@@ -288,13 +317,12 @@ async def process_file(
     ) -> FileSchema:
         """Process a File."""
         file_id = UUID(file_id_str)
-        logger.info(file_id)
+        request.logger.info(file_id)
```
Comment (cappuccinocosmico) [inline backend/routing/file_controller.py:320]: Why does the logger change in this method but nowhere else?

**Pull request merged**